src/lib/scheduler/node_queue_scheduler.cpp

```diff
@@ -209,4 +219,41 @@ void NodeQueueScheduler::_group_tasks(const std::vector<std::shared_ptr<Abstract
   }
 }
 
+// with 240 groups.................
+void NodeQueueScheduler::_group_numa_aware(const std::vector<std::shared_ptr<AbstractTask>>& tasks) const {
+  auto round_robin_counter = std::vector<int>(_queue_count, 0);
+
+  // std::vector<std::shared_ptr<AbstractTask>> grouped_tasks(_workers_per_node * _queue_count);
+  std::vector<std::shared_ptr<AbstractTask>> grouped_tasks(_group_number_per_node * _queue_count);
+
+  for (const auto& task : tasks) {
+    if (!task->predecessors().empty() || !task->successors().empty()) {
+      return;
+    }
+    auto num_node = task->node_id();
+    const auto group_id =
+        (_group_number_per_node * num_node) + (round_robin_counter[num_node] % _group_number_per_node);
```

Comment:
If you make `_group_number_per_node` a compile time constant, you could change the type of `grouped_tasks` to `std::vector<std::array<..., _group_number_per_node>>` and access it as `grouped_tasks[numa_node][round_robin_counter[numa_node]]`